OpenRA.Mods.Common/Traits/Mobile.cs

```diff
@@ -71,6 +71,8 @@ public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
 			else if (locomotorInfos.Count(li => li.Name == Locomotor) > 1)
 				throw new YamlException("There is more than one locomotor named '{0}'.".F(Locomotor));
 
+			locomotor = null;
```

Comment:
Can we please have a comment here explaining why this is needed? At first look this seems to be redundant, but your comments in IRC suggest this is fixing a regression - I guess we are leaking state between worlds with the `ModData.DefaultRules`?